test/Constraints/rdar44816848.swift

```diff
@@ -0,0 +1,13 @@
+// RUN: %target-typecheck-verify-swift
+
+class A {}
+class B: A {}
+class C: A {}
+
+struct S {
+  func foo<T: A>(types: [T.Type]) {}
+}
+
+func bar(_ s: S!) {
+  s.foo(types: [A.self, B.self])
```

Comment:
Nitpick: Since the bindings might not always be tried in the same order, having a second call with the array reversed seems like good future-proofing.

Reply:
I don't think element order here matters match because it used to binding array type to `A` and then equate to `B`, and reverse it would lead to the same result...

Reply:
I went ahead and added both reverse cases to `S!` and `S` just in case :)